### ISSUE TITLE
get provider version to select box in vagrantfile

### ISSUE THREAD

**Issue by ddavisgso** [mitchellh/vagrant#6910]
I tried searching for other answers on this and so forgive if there's a simple way to do this.

Instead of installing a plugin that manages the virtual box guest additions based on the provider available on the host, I'd like the vagrantfile to select an appropriate box which has the correct guest additions already based on the virtualbox provider installed. 

I can do this based on the vagrant version:
box_template=( Vagrant::VERSION =~ /^1.8/ ? "sandbox5" : "sandbox4" )

Actually using the above to determine if can use linked clones, but currently making the general assumption that if a member of my team has upgraded vagrant to 1.8, then they have likely upgraded to virtual box 5.

Can I do something similar from the vagrantfile based on the available virtualBox version installed?  Is the provider version exposed outside of the provider plugin itself?

Thanks,

**Comment by sethvargo** [comment]
Hi @ddavisgso 

Unfortunately there is no direct API for doing something like this. You could use plain Ruby to shell out to the vboxmanage command as part of the Vagrantfile, like this (untested)

```ruby
vbox_version = `VBoxManager --version`.strip

Vagrant.configure(2) do |config|
  if vbox_version =~ /5\.0/
    # ...
  end
end
```

Does that help?

**Comment by ddavisgso** [comment]
I'll test that out.  Seems simple enough and will likely work for this.

Thanks and I'll let you know tomorrow.

**Comment by ddavisgso** [comment]
ok, works for me for now, but I hate what I did to get it working...

vbox_version = `/Progra~1/Oracle/VirtualBox/VBoxManage --version`.strip
box_template=( vbox_version =~ /5\.0/ ? "sandbox5" : "sandbox4" )

I'm on windows and neither the install of virtualbox or vagrant put virtualbox in the %PATH%.  I'm not sure how vagrant finds the provider binaries, but I had to do that junk above.

problem is that won't work for anyone who installed virtualbox differently than I did or runs on a different platform than I do.  Is there a simple way to get the vboxmanager path like vagrant must do elsewhere?

Thanks,
